Updating question - How would you prevent a component from rendering? (#10)
Pau1fitz/react-interview: readme.md

you can use an arrow function in the callback. The problem here is that a new callback is created each time the component renders.

#### How would you prevent a component from rendering?

> Returning null from a component's render method hide the component, but it does not affect the firing of the component's lifecycle methods. 

Reviewer: ```suggestion
Returning null from a component's render method means nothing will be displayed, but it does not affect the firing of the component's lifecycle methods. 
```

> If the amount of the component re-rendering is an issue,  there are two options available. Manually implementing a check in shouldComponentUpdate lifecycle method hook. 

Reviewer: ```suggestion
If the amount of times the component re-renders is an issue, there are two options available. Manually implementing a check in `shouldComponentUpdate` lifecycle method hook. 
```

```
shouldComponentUpdate(nextProps, nextState){
  // Do some check here
  return resultOFCheckAsBoolean
}
```
Or using React.PureComponent instead of React.Component

> React.PureComponent implements shouldComponentUpdate() with a shallow prop and state comparison. So you can avoid re-rendering the component with the same props and state.

Reviewer: ```suggestion
React.PureComponent implements `shouldComponentUpdate()` with a shallow prop and state comparison. This enables you to avoid re-rendering the component with the same props and state.
```
